tutorials/polkadot-sdk/build-a-blockchain/add-trusted-nodes.md

```diff
@@ -0,0 +1,313 @@
+            [
+              "5GdFMFbXy24uz8mFZroFUgdBkY2pq6igBNGAq9tsBfEZRSzP",
+              1
+            ],
+            [
+              "5DJRQQWEaJart5yQnA6gnKLYKHLdpX6V4vHgzAYfNPT2NNuW",
+              1
+            ]
+          ]
+        },
+        ```
+
+5. Save and close `customSpec.json`
+
+### Convert Chain Specification to Raw Format
+
+After creating your custom chain specification, the next crucial step is converting it to a raw format. This process is essential because the raw format includes encoded storage keys that nodes use to reference data in their local storage. By distributing a raw chain specification, you ensure that each node in the network stores data using the same storage keys, which is vital for maintaining data integrity and facilitating network synchronization.
+
+To convert your chain specification to the raw format, follow these steps:
+
+1. Navigate to the root directory where you compiled the node template
+   
+2. Run the following command to convert the `customSpec.json` chain specification to the raw format and save it as `customSpecRaw.json`:
+
+    ```bash
+    ./target/release/solochain-template-node build-spec \
+    --chain=customSpec.json \
+    --raw \
+    --disable-default-bootnode > customSpecRaw.json
+    ```
+
+## Add Keys to the Keystore
+
+To enable block production and finalization, you need to add two types of keys to the keystore for each node in the network:
+
+- `aura` authority keys for block production
+- `grandpa` authority keys for block finalization
+
+Follow these steps for each node in your network:
+
+1. Open a terminal and navigate to the root directory where you compiled the node template
+
+2. Insert the `aura` secret key:
+
+    ```bash
+    ./target/release/solochain-template-node key insert \
+    --base-path /tmp/node01 \
+    --chain customSpecRaw.json \
+    --scheme Sr25519 \
+    --suri "INSERT_SECRET_PHRASE" \
+    --password-interactive \
+    --key-type aura
+    ```
+
+    !!!note
+        Replace `INSERT_SECRET_PHRASE` with the secret phrase or seed you generated earlier. When prompted, enter the password you used to generate the keys.
+
+3. Insert the `grandpa` secret key:
+    ```bash
+    ./target/release/solochain-template-node key insert \
+    --base-path /tmp/node01 \
+    --chain customSpecRaw.json \
+    --scheme Ed25519 \
+    --suri "INSERT_SECRET_PHRASE" \
+    --password-interactive \
+    --key-type gran
+    ```
+
+    !!!note
+        Use the same secret phrase or seed and password as in step 2.
+
+4. Verify that your keys are in the keystore by running the following command:
+
+    ```bash
+    ls /tmp/node01/chains/local_testnet/keystore
+    ```
+
+    You should see output similar to:
+
+    <div id='termynal' data-termynal>
+        <span data-ty>61757261ea23fa399c6bd91af3d7ea2d0ad46c48aff818b285342d9aaf15b3172270e914</span>
+        <span data-ty>6772616ec9c2cd111f98f2bf78bab6787449fc007dd7f2a5d02f099919f7fb50ade97dd6</span>
+    </div>
```

Comment:
I've noticed that the commands haven't been added to these terminal elements, and I've kinda been like ehhh okayyyy, but here in particular looks weird without having the command there 🤷‍♀️